### PR TITLE
fix(trade): fix UnsupportedToken state displaying conditions

### LINKED ISSUE #2799: [1.42]Don't show 'Unsupported wallet' error when not connected
Discusstion https://cowservices.slack.com/archives/C0361CDG8GP/p1688578373834849
Reported in https://github.com/cowprotocol/cowswap/pull/2789#issuecomment-1621426787

**To reproduce:**

1. Switch to gchain being not connected to a wallet
2. Open Advanced orders tab
3. Pick AST token (unsupported) in the To field
4. Select an amount to trade

**AR**: Unsupported wallet error is displayed, even though I’m not even connected.

**ER**: Unsupported token error should be displayed (when not connected)

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2799

image

### CODE REVIEW

**Review by alfetopito — APPROVED** [review]

**Review by elena-zh — COMMENTED** [review]
Hey @shoom3301 , we tend to show Unsupported token warning to a not connected user. 
So SWAP form works this way rn
swap form
Limit orders/TWAP orders don't show this warning now
limit form

I opt for using the same logic everywhere. 

Also, I noticed a similar case on the Advanced orders page when 'Sell amount is too small' error happens: 
image


OR not liquidity
image

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed.
I also removed excessive quote requests when an unsupported token is selected

**Review by elena-zh — APPROVED** [review]
LGTM now, thank you!

File: src/utils/getIsMessageIncludeToken.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export function getIsMessageIncludeToken(message: string, tokenAddress: string): string | null {
```
Comment (alfetopito) [inline src/utils/getIsMessageIncludeToken.ts:1]:
```suggestion
export function getDoesMessageIncludeToken(message: string, tokenAddress: string): string | null {
```

Comment (shoom3301) [reply src/utils/getIsMessageIncludeToken.ts]: Fixed, thanks!

**Pull request merged**